cachi2/core/package_managers/gomod.py

```diff
@@ -718,6 +719,57 @@ def _find_missing_gomod_files(source_path: RootedPath, subpaths: list[str]) -> l
     return invalid_gomod_files
 
 
+def _setup_go_toolchain(go_mod_file: RootedPath, env: Dict) -> Go:
+    go = Go()
+    go_max_version = version.Version("1.21")
+    go_base_version = go.version
+    go_mod_version_msg = "go.mod recommends/requires Go version: {}"
+
+    if (_ := _get_gomod_version(go_mod_file)) is None:
+        # Go added the 'go' directive to go.mod in 1.12 [1]. If missing, 1.16 is assumed [2].
+        # For our version comparison purposes we set the version explicitly to 1.20 if missing.
+        # [1] https://go.dev/doc/go1.12#modules
+        # [2] https://go.dev/ref/mod#go-mod-file-go
+        _ = "1.20"
+        go_mod_version_msg += " " + "(cachi2 enforced)"
+
+    go_mod_version = version.Version(_)
+
+    if go_mod_version.major > go_max_version.major or go_mod_version.minor > go_max_version.minor:
+        raise PackageManagerError(
+            f"Go version '{go_mod_version}' is not supported yet.",
+            solution=(
+                "Please try downgrading your Go version and retry the request. "
+                "You may also want to open a feature request on adding support for this version."
+            ),
+        )
+
+    log.info(go_mod_version_msg.format(go_mod_version))
+
+    if go_mod_version >= go_max_version:
+        if go_base_version < go_max_version:
+            # our base Go installation is too old and we need a newer one to support new keywords
+            go = Go(release="go1.21.0")
+        elif go_base_version > go_mod_version:
+            # we have a newer toolchain than the requirement, so we don't want GOTOOLCHAIN=auto
+            env["GOTOOLCHAIN"] = "local"
```

Comment:
Is this set just for the prefetch? I think we'll need to provide it in the request output

Reply:
darn, yep, good eyes!

Reply:
In fact, this shouldn't even happen here, but more changes will be needed for `_fetch_gomod_source` to be able to make this kind of decision.

Reply:
Based on the discussion in https://github.com/containerbuildsystem/cachi2/issues/501, let's keep `GOTOOLCHAIN=auto` only for prefetch.